src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -115,55 +40,115 @@ const ChatItemMyTextV2 = ({
   message = '',
   status = ChatStatus.PENDING,
   avatar,
-  chatType
+  chatType,
+  messageType,
+  data
 }: ChatItemMyTextProps) => {
-  const messageRef = React.useRef<Text>(null);
+  const {setReplyPreview, onContextMenuPressed} = useMessageHook();
+
+  const swipeableRef = React.useRef<Swipeable | null>(null);
+  const [isNewLine, setIsNewLine] = React.useState(true);
+  const bubblePosition = useSharedValue(0);
+
+  const isReply = messageType === 'reply';
+  const isReplyPrompt = messageType === 'reply_prompt';
+  const isShowUserInfo = !isContinuous || isReplyPrompt || isReply;
+
+  const onTextLayout = (event: NativeSyntheticEvent<TextLayoutEventData>) => {
+    const {lines} = event.nativeEvent;
+    const lastLine = lines[lines.length - 1];
+    const lastLineWidth = lastLine?.width;
+    const isCalculatedNewLine = targetLastLineWidth - lastLineWidth < 24;
+    setIsNewLine(isCalculatedNewLine);
+  };
+
+  const animatedBubbleStyle = useAnimatedStyle(() => ({
+    transform: [{translateX: bubblePosition.value}]
+  }));
 
   const renderIcon = React.useCallback(() => {
     if (status === ChatStatus.PENDING)
       return (
         <View style={styles.icon}>
-          <IconChatClockGrey width={12} height={12} />
+          <IconChatClockGrey color={colors.silver} width={12} height={12} />
         </View>
       );
 
     return (
       <View style={styles.icon}>
-        <IconChatCheckMark />
+        <IconChatCheckMark color={colors.silver} />
       </View>
     );
   }, []);
 
+  const contextMenuActions = [
+    {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
+    {title: 'Copy Message', systemIcon: 'square.on.square'},
+    {title: 'Delete Message', systemIcon: 'trash', destructive: true}
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The context menu actions are hardcoded. It would be better to move them to a constants file or make them configurable through props. This will increase the reusability of the component.

```diff
- const contextMenuActions = [
-   {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
-   {title: 'Copy Message', systemIcon: 'square.on.square'},
-   {title: 'Delete Message', systemIcon: 'trash', destructive: true}
- ];
+ const contextMenuActions = props.contextMenuActions || [
+   {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
+   {title: 'Copy Message', systemIcon: 'square.on.square'},
+   {title: 'Delete Message', systemIcon: 'trash', destructive: true}
+ ];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->